workflow_tests/utils/utils.ts

```diff
@@ -63,28 +87,40 @@ function createMockStep(name, message, job_id = null, inputs = null, in_envs = n
             mockWithCommand += `\necho "${key}=${value}" >> "$GITHUB_OUTPUT"`;
         });
     }
-    if (out_envs) {
-        Object.entries(out_envs).forEach(([key, value]) => {
+    if (outEnvs) {
+        Object.entries(outEnvs).forEach(([key, value]) => {
             mockWithCommand += `\necho "${key}=${value}" >> "$GITHUB_ENV"`;
         });
     }
     if (!isSuccessful) {
         mockWithCommand += '\nexit 1';
     }
-    const mockStep = {
+    if (!id) {
+        return {
+            name: mockStepName,
+            mockWith: mockWithCommand,
+        };
+    }
+
+    return {
+        id,
         name: mockStepName,
         mockWith: mockWithCommand,
     };
-    if (id) {
-        mockStep.id = id;
-    }
-    return mockStep;
 }
 
-function createStepAssertion(name, isSuccessful = true, expectedOutput = null, jobId = null, message = null, inputs = null, envs = null) {
+function createStepAssertion(
+    name: string,
+    isSuccessful = true,
+    expectedOutput = null,
+    jobId: string | null = null,
+    message: string | null = null,
+    inputs: Array<{key: string; value: string}> | null = null,
+    envs: Array<{key: string; value: string}> | null = null,
```

Comment:
Same
```suggestion
    jobId?: string,
    message?: string,
    inputs?: Array<{key: string; value: string}>,
    envs?: Array<{key: string; value: string}>,
```
Also maybe you can create a separate type for `{key: string; value: string}`?

Reply:
Separate type for `{key: string; value: string}` - sure I'll apply that. I was talking about this with @blazejkustra, because it seemed quite weird to do it this way instead of just using records + Object.entries, but regarding the first part just like in one of the comments above - probably a separate PR.